File: ruby/test/solutions/0217_contains_duplicate/contains_duplicate_test.rb
# frozen_string_literal: true

require 'test_helper'
require_relative '../../../lib/solutions/0217_contains_duplicate/contains_duplicate'

class ContainsDuplicateTest < Minitest::Test
  def test_case_1_works
    assert contains_duplicate([1, 2, 3, 1])
  end

  def test_case_2_works
    refute contains_duplicate([1, 2, 3, 4])
  end

  def test_case_3_works
    assert contains_duplicate([1, 1, 1, 3, 3, 4, 3, 2, 4, 2])
  end
end

File: elixir/test/solutions/0217_contains_duplicate/contains_duplicate_test.exs
defmodule LeetCode.Solutions.ContainsDuplicateTest do
  use ExUnit.Case, async: true

  alias LeetCode.Solutions.ContainsDuplicate

  test "Case 1 works" do
    assert ContainsDuplicate.call([1, 2, 3, 1]) == true
  end

  test "Case 2 works" do
    assert ContainsDuplicate.call([1, 2, 3, 4]) == false
  end

  test "Case 3 works" do
    assert ContainsDuplicate.call([1, 1, 1, 3, 3, 4, 3, 2, 4, 2]) == true
  end

  test "Edge Case 1 works" do
    assert ContainsDuplicate.call([3, 3]) == true
  end

  test "Edge Case 2 works" do
    assert ContainsDuplicate.call([0, 4, 5, 0, 3, 6]) == true
  end
end
